### PR TITLE
WiP: BK+TST: add_archive_content exclude bug

### PR DESCRIPTION
**Pull request by debanjum**
Refer issue #735

### CODE REVIEW

File: datalad/interface/add_archive_content.py [hunk 1]
```diff
@@ -323,6 +323,17 @@ def __call__(archive, annex=None,
             stats = ActivityStats()
 
             for extracted_file in earchive.get_extracted_files():
+                # continue to next iteration if extracted_file in excluded
+                if exclude:
```
Comment (yarikoptic) [inline datalad/interface/add_archive_content.py]: Well I guess the idea was to apply exclude after filenames got tuned up - striped, renamed, etc. Need to think if I used that assumption somewhere already

File: datalad/interface/tests/test_add_archive_content.py [hunk 2]
```diff
@@ -36,6 +37,56 @@
...
+                        delete=True,
+                        leading_dirs_consider=['crcns.*', '1'],
+                        leading_dirs_depth=2,
+                        use_current_dir=False,
+                        exclude='.*__MACOSX.*')  # some junk penetrates
+
+    # regression test: the subdir in MACOSX wasn't excluded and its name was getting stripped by leading_dir_len
+    assert_false(exists('__MACOSX'))  # if stripping and exclude didn't work this fails
+    assert_false(exists('c-2_data'))  # if exclude doesn't work then name of subdir gets stripped by leading_dir_len
+    assert_false(exists('CR24B'))     # if exclude doesn't work but everything else works this fails
```
Comment (yarikoptic) [inline datalad/interface/tests/test_add_archive_content.py]: Use find_files and compare full set

File: datalad/interface/add_archive_content.py (hunk 1 above)
Comment (yarikoptic) [inline datalad/interface/add_archive_content.py]: ok, I can live with this solution I think ;)  just move it after `stats.files += 1` line

**Comment by yarikoptic** [comment]
suboptimal (why to delay the check if you don't use modified filename) but ok.  merging

**Comment by glalteva** [comment]
@yarikoptic ahh- was emailing you right now to discuss another possible bug with CRCNS - I thought our solution might be pre-mature and there could be a different fix. I was playing around with it this morning and placed the check back where it was originally before we started. Will send my email in a few - or will you be in today?

**Comment by yarikoptic** [comment]
That is OK
Only email today. Tomorrow will be a proper work day

**Pull request merged**